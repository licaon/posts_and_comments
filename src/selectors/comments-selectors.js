export const getComments = state => state.CommentsReducer;
export const getCommentsState = state => state.CommentsStateReducer;
