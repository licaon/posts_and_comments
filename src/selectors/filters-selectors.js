export const getSearchFiltersValue = state => state.SearchReducer;
