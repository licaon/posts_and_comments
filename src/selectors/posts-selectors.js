import { createSelector } from 'reselect';

export const getPosts = state => state.PostsReducer.posts;

export const getFilteredPosts = createSelector(
  [getPosts],
  posts => posts,
);
