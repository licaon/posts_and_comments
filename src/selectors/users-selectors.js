export const getUsers = state => state.UsersReducer.users;
