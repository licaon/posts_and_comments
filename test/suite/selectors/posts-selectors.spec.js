import * as PostsSelectors from 'src/selectors/posts-selectors';

describe('PostsSelectors', () => {
  it(`should execute getPosts correctly`, (done) => {
    expect(PostsSelectors.getPosts({
      PostsReducer: {
        posts: 'PostsReducer.posts',
      },
    })).toMatchSnapshot();

    done();
  });
  it(`should execute getSelectFiltersValue correctly`, (done) => {
    expect(PostsSelectors.getPostsFilterBySearch.resultFunc(
      [
        {
          field1: 'testing',
          field2: 'something else',
        },
        {
          field1: 'something else',
          field2: 'something else',
        },
        {
          field1: 'something else',
          field2: 'TEST',
        },
      ],
      {
        search: {
          value: 'test',
          fields: ['field1', 'field2'],
        }
      }
    )).toMatchSnapshot();

    done();
  });
  it(`should execute getPostsFilteredBySelect correctly`, (done) => {
    expect(PostsSelectors.getPostsFilteredBySelect.resultFunc(
      [
        {
          field1: 'testing',
          field2: 'something else',
        },
        {
          field1: 'something else',
          field2: 'something else',
        },
        {
          field1: 'test',
          field2: 'something else',
        },
        {
          field1: 'something else',
          field2: 'test',
        },
        {
          field1: 'something else',
          field2: 'TEST',
        },
      ],
      {
        search: {
          value: 'test',
          fields: ['field1', 'field2'],
        },
      },
    )).toMatchSnapshot();

    done();
  });
  it(`should execute getFilteredPosts correctly`, (done) => {
    expect(PostsSelectors.getFilteredPosts.resultFunc(
      [
        {
          id: 'test',
          field: 'something else',
        },
        {
          id: 'something else',
          field: 'something else',
        },
        {
          id: 'another id 2',
          field: 'TEST',
        },
      ],
      [
        {
          id: 'test',
          field: 'something else',
        },
        {
          id: 'something else',
          field: 'test',
        },
        {
          id: 'another id 1',
          field: 'TEST',
        },
      ],
    )).toMatchSnapshot();

    done();
  });
});
