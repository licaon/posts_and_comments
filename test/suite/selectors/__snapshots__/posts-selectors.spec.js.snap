// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsSelectors should execute getFilteredPosts correctly 1`] = `
Array [
  Object {
    "field": "something else",
    "id": "test",
  },
  Object {
    "field": "something else",
    "id": "something else",
  },
]
`;

exports[`PostsSelectors should execute getPosts correctly 1`] = `"PostsReducer.posts"`;

exports[`PostsSelectors should execute getPostsFilteredBySelect correctly 1`] = `
Array [
  Object {
    "field1": "test",
    "field2": "something else",
  },
  Object {
    "field1": "something else",
    "field2": "test",
  },
]
`;

exports[`PostsSelectors should execute getSelectFiltersValue correctly 1`] = `
Array [
  Object {
    "field1": "testing",
    "field2": "something else",
  },
  Object {
    "field1": "something else",
    "field2": "TEST",
  },
]
`;
